src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.Replace.cs

```diff
@@ -186,7 +186,7 @@ private static string Replace(MatchEvaluator evaluator, Regex regex, string inpu
                             vsb.Append(input.AsSpan(prevat, match.Index - prevat));
 
                         prevat = match.Index + match.Length;
-                        vsb.Append(evaluator(match));
+                        vsb.Append(evaluator(match) ?? "");
```

Comment:
You reviewed the other Append call sites and this was the only one that needed the null check?

Reply:
Yes. Most are obvious (char or Span). A couple are not mainly
```
                switch (child.Type())
                {
                    case RegexNode.Multi:
                        vsb.Append(child.Str);
```
After some stepping I don't think I can find a way for `Str` to be null as it originates from succesfully parsing the replacement string. I suppose I could put in a defensive `?? ""` but I think it's OK.

Reply:
It's OK with idle calling of ValueStringBuilder.Append, that have no cases for empty strings ?

Reply:
> It's OK with idle call of ValueStringBuilder.Append, that have no cases for empty strings ?

@ZOXEXIVO, I don't understand your question. Can you rephrase?

Reply:
> @ZOXEXIVO, I don't understand your question. Can you rephrase?
Why we call Append with empty string if evaluator(match) return null ?

Reply:
This could have instead been:
```C#
string result = evaluator(match);
if (result != null)
{
    vsb.Append(result);
}
```
My guess is @danmosemsft felt it wasn't worth the extra verbosity because nulls are very rare and don't show up on hot paths.  But I'm ok either way if he wants to change it.

Reply:
Right, but I'm happy to change it..